tests/bundles/test_index_builder.py

```diff
@@ -24,7 +24,7 @@ def test_index_builder_build_and_push():
     # <registry_url>/v2/ and our quay.io registry doesn't support this workflow
     index_builder = IndexBuilder(docker_api=docker_api, debug=True)
     index_image = index_builder.build_and_push(
-        bundles, hash_string, skip_validation=True
+        bundles, hash_string, skip_validation=True, test_run=True
```

Comment:
it seems like what you really need it to mock the registry.

Reply:
Update on this: We'll take a look at the LocalRegistry setup to see if we can modify it to match our use-case i.e. image's being accessible at `<local-test-registry-url>/v2`.

We have some ideas but we'll validate them tomorrow :) 

cc: @rpriyanshu9

Reply:
Or it can use a local registry, as you mentioned. Personally, I'd go with the github-mirror example, for good-enough-with-less-dependencies, but any strategy is fine really.

Reply:
Thanks a lot Amador for these references. Let's take a look at them to figure out the most convenient approach possible to meet our use-case :)